parsl/tests/test_monitoring/test_htex_init_blocks_vs_monitoring.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import os
+import parsl
+import pytest
+import time
+
+from parsl.providers import LocalProvider
+from parsl.channels import LocalChannel
+from parsl.launchers import SimpleLauncher
+
+from parsl.config import Config
+from parsl.executors import HighThroughputExecutor
+from parsl.monitoring import MonitoringHub
+
+
+def fresh_config(run_dir, strategy):
+    return Config(
+        run_dir=os.fspath(run_dir),
+        executors=[
+            HighThroughputExecutor(
+                label="htex_local",
+                cores_per_worker=1,
+                encrypted=True,
+                provider=LocalProvider(
+                    channel=LocalChannel(),
+                    init_blocks=1,
+                    # min and max are set to 0 to ensure that we don't get
+                    # a block from ongoing strategy scaling, only from
+                    # init_blocks
+                    min_blocks=0,
+                    max_blocks=0,
+                    launcher=SimpleLauncher(),
+                ),
+            )
+        ],
+        strategy=strategy,
+        strategy_period=0.1,
+        monitoring=MonitoringHub(
+                        hub_address="localhost",
+                        hub_port=55055,
+                        logging_endpoint=f"sqlite:///{run_dir}/monitoring.db"
+        )
+    )
+
+
+@parsl.python_app
+def this_app():
+    pass
+
+
+@pytest.mark.local
+@pytest.mark.parametrize("strategy", ('none', 'simple', 'htex_auto_scale'))
+def test_row_counts(tmpd_cwd, strategy):
+    # this is imported here rather than at module level because
+    # it isn't available in a plain parsl install, so this module
+    # would otherwise fail to import and break even a basic test
+    # run.
+    import sqlalchemy
+    from sqlalchemy import text
+
+    parsl.load(fresh_config(tmpd_cwd, strategy))
+
+    this_app().result()
+
+    parsl.dfk().cleanup()
+    parsl.clear()
+
+    # at this point, we should find one row in the monitoring database.
+
+    engine = sqlalchemy.create_engine("sqlite:///runinfo/monitoring.db")
+    with engine.begin() as connection:
+
+        # we should see a single block:
+        result = connection.execute(text("SELECT COUNT(DISTINCT block_id) FROM block"))
+        (c, ) = result.first()
+        assert c == 1
```

Comment:
This is a slightly subtle "what's going on here?" assertion.  For these, I often write something to communicate the assertion's purpose.  In this case, this feels to me like a "test setup verification" in that it's verifying that something larger isn't in play (e.g., the sqlite DB being shared between tests and not otherwise cleaned up).  Consider:
```python
    assert c == 1, "Verify test setup"
```